src/lib/ingest.js

```diff
@@ -55,75 +54,74 @@ export async function convertIngestObjectToDbObject(
   return {
     index,
     id: dbDataObject.id,
-    action: 'index',
+    action,
     _retry_on_conflict: 3,
     body: dbDataObject
   }
 }
 
-export function combineDbObjectsIntoBulkOperations(records) {
-  const operations = records.reduce((/** @type {{}[]} */ bulkOperations, record) => {
-    const operation = {}
-    operation[record.action] = {
-      _index: record.index,
-      _type: record.type,
-      _id: record.id
-    }
-    if (record.parent) {
-      operation[record.action]._parent = record.parent
-    }
+export async function executeDbOperation(
+  /** @type {{ index: string; id: string; body: {}; action: string}} */ record
+) {
+  const { index, id, body, action } = record
 
-    bulkOperations.push(operation)
-    if (record.action !== 'delete') {
-      bulkOperations.push(record.body)
-    }
-    return bulkOperations
-  }, [])
-  return operations
-}
+  if (!index) {
```

Comment:
all operations require an index -- maybe in the future, we'll have some that don't.